admin/src/App.js

```diff
@@ -16,14 +16,19 @@ import ScoreboardManager from "./components/ScoreboardManager";
 import BackendLog from "./components/BackendLog";
 import Dashboard from "./components/Dashboard";
 import Analytics from "./components/AnalyticsEvent";
+import { localStorageGet, localStorageSet } from "./utils";
 
 function App() {
-    const [isOverlayPreviewShown, setIsOverlayPreviewShown] = useState(false);
+    const [isOverlayPreviewShown, setIsOverlayPreviewShown] = useState(localStorageGet("OverlayPreviewShown"));
```

Comment:
Instead of this huge boilerplate you could instead create a single hook which automatically saves state in the localstorage and acts like regular useState
Something like useStorageState